src/ir/proto/predicate.cc

```diff
@@ -0,0 +1,94 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+#include "src/ir/proto/predicate.h"
+
+#include <memory>
+
+#include "src/ir/predicates/predicate.h"
+
+namespace raksha::ir::proto {
+
+namespace ir = raksha::ir;
+namespace pred = raksha::ir::predicates;
+
+std::unique_ptr<pred::Predicate> Decode(
+    const arcs::InformationFlowLabelProto_Predicate &predicate_proto) {
+  switch (predicate_proto.predicate_case()) {
+    case arcs::InformationFlowLabelProto_Predicate::kLabel: {
+      const arcs::InformationFlowLabelProto &label = predicate_proto.label();
```

Comment:
`const auto &label`?

Reply:
In general, I erred towards writing out the exact type rather than using type inference, as it seems like that's what the Google C++ style guide prefers. See https://google.github.io/styleguide/cppguide.html#Type_deduction .

Reply:
SG. It is fine to have the type explicitly specified. However, `const auto &` would have been OK here. The case label `arcs::InformationFlowLabelProto_Predicate::kLabel` already provides an indication of what the type is. Therefore, it doesn't hinder readability. In fact, I think it would have improved readability by making the code concise.